### ISSUE TITLE
NPE in CmsWorkplaceMessages.java:144

### ISSUE THREAD

**Issue by gallardo** [alkacon/opencms-core#573]
Found in one of our test environments:

```
01-Feb-2018 10:53:38.872 SEVERE [http-nio-8080-exec-6] org.apache.catalina.core.ApplicationContext.log OpenCmsServlet: Error while adding resource-type {0} to gallery type list.
 java.lang.NullPointerException
	at org.opencms.workplace.CmsWorkplaceMessages.getResourceTypeDescription(CmsWorkplaceMessages.java:144)
	at org.opencms.ade.galleries.CmsGalleryService.createTypeBean(CmsGalleryService.java:1945)
	at org.opencms.ade.galleries.CmsGalleryService.buildTypesList(CmsGalleryService.java:1779)
	at org.opencms.ade.galleries.CmsGalleryService.buildTypesList(CmsGalleryService.java:556)
	at org.opencms.ade.galleries.CmsGalleryService.getResourceTypeBeans(CmsGalleryService.java:2515)
	at org.opencms.ade.galleries.CmsGalleryService.getInitialSettingsInternal(CmsGalleryService.java:2293)
	at org.opencms.ade.galleries.CmsGalleryService.getInitialSettings(CmsGalleryService.java:784)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.opencms.gwt.CmsGwtService.service(CmsGwtService.java:194)
	at org.opencms.main.OpenCmsCore.invokeGwtService(OpenCmsCore.java:1719)
	at org.opencms.main.OpenCmsServlet.doGet(OpenCmsServlet.java:158)
	at org.opencms.main.OpenCmsServlet.doPost(OpenCmsServlet.java:177)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.opencms.jsp.jsonpart.CmsJsonPartFilter.doFilter(CmsJsonPartFilter.java:281)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:716)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:466)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:391)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:318)
	at org.opencms.main.OpenCmsUrlServletFilter.doFilter(OpenCmsUrlServletFilter.java:127)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:94)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:504)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.StuckThreadDetectionValve.invoke(StuckThreadDetectionValve.java:206)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:620)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:502)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1132)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1533)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1489)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by tHerrmann** [comment]
This will happen in case a resource type is referenced, where no explorer type is configured.
Check your configuration to avoid this issue.
We will improve the handling there, to avoid this null pointer in future OpenCms versions.

**Comment by gallardo** [comment]
Great! :+1:

Alberto

2018-04-16 11:37 GMT+02:00 Tobias Herrmann <notifications@github.com>:

**Comment by tHerrmann** [comment]
Fixed with commit c89aa936f2eb556eb6cd3bfee1aa1e3c10297ee5.

**Comment by tHerrmann** [comment]
This has been fixed with release 10.5.4.
